Source/WTF/wtf/text/ASCIILiteral.h

```diff
@@ -136,26 +138,25 @@ inline ASCIILiteral ASCIILiteral::deletedValue()
 
 inline namespace StringLiterals {
 
-constexpr ASCIILiteral operator"" _s(const char* characters, size_t)
+constexpr ASCIILiteral operator"" _s(const char* characters, size_t n)
 {
-    auto result = ASCIILiteral::fromLiteralUnsafe(characters);
+    auto result = ASCIILiteral::fromLiteralUnsafe(characters, n);
     ASSERT_UNDER_CONSTEXPR_CONTEXT(result.characters() == characters);
     return result;
 }
```

Comment:
LOL WUT.

```
$ ./Tools/Scripts/run-api-tests --release --no-build TestWTF.WTF.StringViewEqualBasic 
Running tests       
                  
./Tools/TestWebKitAPI/Tests/WTF/StringView.cpp:327
Value of: a == "Hell\0"_s
  Actual: true
Expected: false



./Tools/TestWebKitAPI/Tests/WTF/StringView.cpp:331
Value of: test3 == "Hello\0"_s
  Actual: false
Expected: true


worker/0 TestWTF.WTF.StringViewEqualBasic Failed
Ran 1 tests of 1 with 0 successful
------------------------------
Test suite failed

Failed

    TestWTF.WTF.StringViewEqualBasic
        
        ./Tools/TestWebKitAPI/Tests/WTF/StringView.cpp:327
        Value of: a == "Hell\0"_s
          Actual: true
        Expected: false
        
        
        
        ./Tools/TestWebKitAPI/Tests/WTF/StringView.cpp:331
        Value of: test3 == "Hello\0"_s
          Actual: false
        Expected: true
```

```
$ ./Tools/Scripts/run-api-tests --release --no-build TestWebKitAPI.ParsedContentType.Serialize
Running tests       

./Tools/TestWebKitAPI/Tests/WebCore/ParsedContentType.cpp:196
Expected equality of these values:
  serializeIfValid("text/plain\0"_s)
    Which is: "NOTVALID"
  "text/plain"


worker/0 TestWebKitAPI.ParsedContentType.Serialize Failed
Ran 1 tests of 1 with 0 successful
------------------------------
Test suite failed

Failed

    TestWebKitAPI.ParsedContentType.Serialize
        
        ./Tools/TestWebKitAPI/Tests/WebCore/ParsedContentType.cpp:196
        Expected equality of these values:
          serializeIfValid("text/plain\0"_s)
            Which is: "NOTVALID"
          "text/plain"
```

Note that `_span` doesn't have this behavior currently @geoffreygaren.

We're paying the runtime cost of `strlen()` for every `_s` operator to catch programmer errors.

Is there a fancy, template-y way to catch the `\0` in the string at build time?

Reply:
> Is there a fancy, template-y way to catch the \0 in the string at build time?

I figured out how to make one of these.  See [User-defined literals](https://en.cppreference.com/w/cpp/language/user_literal).  I'll include it in another patch.

Reply:
Turns out user-defined literals treat the global C-string as an array of bytes (passed around via function argument or template parameter) , but they never allocate the C-string as a global C-string, so they can't be used to check that the C-string itself is well-formed (no NULL bytes and all `isASCII()` characters).

It's too bad there's no way in C++ to get both the C-string global and the user-defined literal, because the literal could be used for validation (with a `static_assert()` expression, and the global could be used for the actual bytes.